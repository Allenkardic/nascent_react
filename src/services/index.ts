// export const { useGetPokemonByNameQuery } = pokemonApi;
export { default as pokemonApi } from './pokemon';
