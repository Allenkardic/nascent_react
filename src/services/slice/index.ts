export * from './pokemonSlice';
export * from './singlePokemonSlice';
export * from './savedPokemonSlice';
