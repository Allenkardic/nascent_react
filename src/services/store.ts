import { configureStore, Action } from '@reduxjs/toolkit';
import { useDispatch, useSelector } from 'react-redux';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { ThunkAction } from 'redux-thunk';

import rootReducer, { RootState } from './root-reducer';

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['pokemon'],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      serializableCheck: false,
      immutableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector = () => useSelector;

export type AppThunk = ThunkAction<void, RootState, unknown, Action<string>>;
export const persistor = persistStore(store);
