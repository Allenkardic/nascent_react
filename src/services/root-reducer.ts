import { combineReducers } from '@reduxjs/toolkit';

import { pokemonSliceReducer, singlPokemonSliceReducer, savedPokemonSliceReducer } from './slice';

const rootReducer = combineReducers({
  pokemon: pokemonSliceReducer,
  singlePokemon: singlPokemonSliceReducer,
  savedPokemon: savedPokemonSliceReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
