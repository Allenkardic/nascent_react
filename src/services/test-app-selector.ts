import { pokemonSliceReducer, singlPokemonSliceReducer, savedPokemonSliceReducer } from './slice';
const state = {
  pokemon: pokemonSliceReducer,
  singlePokemon: singlPokemonSliceReducer,
  savedPokemon: savedPokemonSliceReducer,
};

export const testAppSelector = (f: any) => f(state);
