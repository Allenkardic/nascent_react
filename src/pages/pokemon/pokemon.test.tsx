import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { Pokemon } from '..';
import { store } from '../../services';
import 'jest-styled-components';

test('it renders correctly', () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Pokemon />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
