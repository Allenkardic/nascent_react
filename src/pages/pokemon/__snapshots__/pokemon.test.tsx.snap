// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders Pokeman page correctly 1`] = `
.c9 {
  padding-top: 16px;
  background-color: #FFFFFF;
  display: flex;
  flex-direction: column;
  margin-left: auto;
  margin-right: auto;
  padding-bottom: 150px;
}

.c10 {
  padding: 0px 16px;
}

.c1 {
  font-weight: 400;
  color: #333333;
  font-size: 15px;
  text-align: left;
  font-family: Medium;
}

.c6 {
  font-weight: 700;
  color: #090E29;
  font-size: 13px;
  text-align: center;
  font-family: Medium;
}

.c8 {
  font-weight: 500;
  color: #9293a0;
  font-size: 13px;
  text-align: center;
  font-family: Medium;
}

.c0 {
  display: flex;
  display: webkit-flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100px;
  padding: 0px 160px;
  border-bottom: 1px solid #E9EBF5;
  background-color: transparent;
}

.c2 {
  margin-top: 32px;
}

.c4 {
  width: 80px;
}

.c5 {
  height: 4px;
  border-radius: 8px;
  background-color: #2A3362;
  margin-bottom: 4px;
}

.c7 {
  height: 4px;
  border-radius: 8px;
  background-color: #EEEEEE;
  margin-bottom: 4px;
}

.c3 {
  display: flex;
  display: webkit-flex;
  flex-direction: row;
  align-items: center;
  align-items: webkit-align-center;
  justify-content: center;
  column-gap: 16px;
}

@media only screen and (min-width: 768px) {
  .c10 {
    width: 100vw;
    min-width: 100vw;
    margin-left: auto;
    margin-right: auto;
  }
}

@media (min-width: 48rem) {
  .c1 {
    font-size: 15px;
  }
}

@media (min-width: 48rem) {

}

@media (min-width: 48rem) {

}

@media (min-width: 48rem) {

}

@media (min-width: 48rem) {
  .c6 {
    font-size: 13px;
  }
}

@media (min-width: 48rem) {
  .c8 {
    font-size: 13px;
  }
}

@media (min-width: 48rem) {

}

@media (max-width: 768px) {
  .c0 {
    padding: 0px 16px;
    height: 74px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Select Pokemon
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          />
          <div
            class="c6"
            color="#090E29"
            fw="700"
            ta="center"
          >
            User details
          </div>
        </div>
        <div
          class="c4"
        >
          <div
            class="c5"
          />
          <div
            class="c6"
            color="#090E29"
            fw="700"
            ta="center"
          >
            Pokemon
          </div>
        </div>
        <div
          class="c4"
        >
          <div
            class="c7"
          />
          <div
            class="c8"
            color="#9293a0"
            fw="500"
            ta="center"
          >
            Review
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="c9"
    >
      <div
        class="c10"
      >
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
    </div>
  </div>
</div>
`;
